remote-view/shared/src/main/scala/parsley/debug/RemoteView.scala

```diff
@@ -206,23 +193,37 @@ object RemoteView extends DebugView.Reusable with RemoteView {
     
     addrLenValid && addrDotValid && addrNumValid 
   }
-  
+
   /** Create a new instance of [[RemoteView]] with a given custom port. */
-  def apply(userPort: Integer = port, userAddress: String = address): RemoteView = new RemoteView {
-    require(userPort <= MaxUserPort, s"Remote View port invalid : $userPort > $MaxUserPort")
-    require(checkIp(userAddress), s"Remote View address invalid : $userAddress")
-    
-    override protected val port = userPort
-    override protected val address = userAddress
+  def apply(port: Integer = defaultPort, address: String = defaultAddress): RemoteView = {
+    require(port <= MaxUserPort, s"Remote View port invalid : $port > $MaxUserPort")
+    require(checkIp(address), s"Remote View address invalid : $address")
+    new RemoteViewer(port, address)
   }
+  
+  /**
+  * Default number of breakpoints skipped.
+  * 
+  * This value will be returned if
+  * - RemoteView cannot connect to the specified server.
+  * - Server does not actually return a value.
+  * 
+  * -1 represents that the parser should stop immediately. This is so 
+  * that if the user is debugging infinite recursion, the lack of a valid
+  * server will not cause the user's machine to burst into flames.
+  */
+  private [debug] final val DefaultBreakpointSkip = -1
 }
 
 /** Helper object for connecting to the DILL backend. */
-object DillRemoteView extends DebugView.Reusable with RemoteView {
+object DillRemoteView {
   // Default endpoint for DILL backend is port 17484 ("DL") on localhost
-  override protected val port: Int = 17484
-  override protected val address: String = "127.0.0.1"
+  private val port: Int = 17484
+  private val address: String = "127.0.0.1"
+
+  implicit def toViewer(companion: DillRemoteView.type): RemoteView = companion.apply
   
   /** Create a new instance of [[RemoteView]] with default ports for the DILL backend server. */
-  def apply(userAddress: String = address): RemoteView = RemoteView(port, userAddress)
+  def apply: RemoteView = RemoteView(port, address)
```

Comment:
I would probably avoid `apply` for this, because it has no arguments. Perhaps we can just move to `RemoteView.dill` and `RemoteView.dill(userAddress: String)`.

Reply:
Also, I'd explicitly call `apply(address)` here.